### ISSUE TITLE
Citations

### ISSUE THREAD

**Issue by edwardabraham** [scholmd/scholmd#6]
Settle on an implementation of citations. Currently the wiki has a couple of proposed formats. Resolve which format is preferred and illustrate its use. As a starting point, perhaps settle on the pandoc-citeproc style, which uses keys, e.g. `[@abraham_issue_5]`, that refer to an external bibtex file, referenced in the YAML header. This should also work with other bibliographic input formats, such as JSON or YAML, and  these should be documented and tested.

The elife example could be refactored to make use of this style.

This issue is raised as being suitable for the July 2014 Mozilla Open Science sprint.

**Comment by cirosantilli** [comment]
Related on Markdown Testsuite: https://github.com/karlcow/markdown-testsuite/issues/53 "fix a single syntax for everything."

**Comment by mfenner** [comment]
Totally agree. Since the initial discussions we seemed to have settled on Pandoc as the default parser for Scholalry Markdown so we can settle on that style.

**Comment by cboettig** [comment]
One limitation of pandoc citations relates to issue #12: in-line citations do not link to the appropriate section of the bibliography (or better, provide `title` attribute link text that would display the full citation on mouse-over).  

Unlike the other intra-link issues, one shouldn't need new markdown syntax to accomplish this -- for instance, most tex flavors will do the automatic linking to the bibliography.  

This could be accomplished using an appropriate stylesheet, but adding links to inline citation styles does not seem possible in the current CSL schema: https://github.com/citation-style-language/documentation/issues/33 and as John McFarlane puts it, "inline links are not as simple to implement as it might seem" https://groups.google.com/forum/#!msg/pandoc-discuss/0UlUwrViCOQ/KBBs-6rNvY0J

**Comment by trvrb** [comment]
I know you've thought about this a lot, but I'm not a huge fan of the external BibTex file.  My proposal is here: https://github.com/trvrb/refdown.

The ideal is to just link to DOIs from within the .md file and then compile these DOIs to a full reference list.  This way the original .md file is still completely usable (click on links to go to articles), but the compiled version is suitable for journal submission.

**Comment by cboettig** [comment]
@trvrb Yeah, I really like that approach too. It's simple, elegant, and even markdown-parser agnostic.  

Nevertheless I still find it difficult to write a paper in which I (a) I have DOIs for every reference, and (b) CrossRef has the correct metadata for each of them.  I use a similar approach in https://github.com/cboettig/knitcitations/ (which looks up the article metadata on CrossRef, Arxiv, HTML metadata/dublin core or pdf extraction), but some sources can still be trouble.  I hope in future that will be less and less of a problem.  

Look forward to checking out your script.  Seems like you would still has the same CSL limitations that I mention above? (e.g. in hyper-linking to the reference list or automatically adding mouse-over text?)

**Comment by trvrb** [comment]
@cboettig ---

I don't user Knitr, so I hadn't seen knitcitations, but it looks very cleverly done.  I see how cite by URL is absolutely necessary and arXiv is super handy.  I'll look more into it.  Thanks!

**Comment by mfenner** [comment]
@trvrb your document looks really nice. @cboettig and I had a discussion about similar ideas last year: http://blog.martinfenner.org/2013/06/19/citations-in-scholarly-markdown/
